adyenv6b2ccheckoutaddon/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -534,16 +540,47 @@ public void handlePaymentForm(AdyenPaymentForm adyenPaymentForm, BindingResult b
             return;
         }
 
+        CustomerModel customer = handleGuestCustomer(cartModel);
+
         //Update CartModel
         cartModel.setAdyenCseToken(adyenPaymentForm.getCseToken());
         cartModel.setAdyenDfValue(adyenPaymentForm.getDfValue());
+        cartModel.setUser(customer);
+        recalculateCart(cartModel);
 
         //Create payment info
         PaymentInfoModel paymentInfo = createPaymentInfo(cartModel, adyenPaymentForm);
         cartModel.setPaymentInfo(paymentInfo);
+
         modelService.save(cartModel);
     }
 
+    private CustomerModel handleGuestCustomer(CartModel cartModel) {
+        CustomerModel customer = getCheckoutCustomerStrategy().getCurrentUserForCheckout();
```

Comment:
So problem this is fixing is that guest name is not transferred to the order?

Where does Hybris out-of-the-box do this usually for guests and why didn't it get invoked with the Adyen AddOn installed?